content/exchange/artifacts/Linux.Forensics.BodyFile.yaml

```diff
@@ -0,0 +1,67 @@
+name: Linux.Forensics.BodyFile
+author: Chris DiSalle - @chrisdfir
+description: |
+  This artifact collects detailed file metadata from Linux directories specified by the TargetDirectories parameter. The default directories include /tmp, /home, /etc, /var, /usr, /boot, /opt, /lib, and /root, but users can modify this list as needed.
+  Metadata such as file size, modification time, access time, creation time, and file type (directory or symbolic link) is gathered recursively up to a depth defined by the MaxRecursionDepth parameter.
+  For symbolic links, the artifact also captures the target path that the symlink points to. 
+  
+reference:
+  - https://trustedsec.com/blog/incident-response-bring-out-the-body-file
+  - https://wiki.sleuthkit.org/index.php?title=Body_file
+
+precondition: SELECT OS FROM info() WHERE OS = 'linux'
+
+parameters:
+  - name: TargetDirectories
+    type: csv
+    default: |
+      Glob
+      /tmp
+      /home
+      /etc
+      /var
+      /usr
+      /boot
+      /opt
+      /lib
+      /root
+      
+  - name: MaxRecursionDepth
+    type: int
+    default: 3
+    description: The maximum depth of recursion for directories.
+
+sources:
+  - name: BodyFileStats
+    query: |
+      -- Collect file paths from the target directories using glob
+      LET FilePaths = SELECT OSPath FROM glob(globs=TargetDirectories.Glob)
+      
+      -- Use foreach to iterate over the directories and run find recursively
+      LET RecurseFiles = SELECT Stdout FROM foreach(row=FilePaths, query={
+        SELECT * FROM execve(argv=["find", str(OSPath), "-maxdepth", MaxRecursionDepth])
```

Comment:
There is probably no point shelling out to find - glob is quite capable of doing this.

 Can you please explain how this is different from the usual file finder?

Reply:
Understood. I had difficulties getting a recursion parameter to work correctly using other methods. While not ideal to call out to find, I found this method to work quickly and effectively. Open to your advice.

Reply:
Take a look at the file finder https://docs.velociraptor.app/artifact_references/pages/linux.search.filefinder/ which does pretty much what this artifact is supposed to do - to get recursion use the double star glob `**` as described here https://docs.velociraptor.app/docs/forensic/filesystem/

Reply:
Thanks for the reference. I will take a look at the adjustment as time permits.